lib/inky/inky.ex

```diff
@@ -0,0 +1,625 @@
+    send_command(state, 0x3B, 0x04)
+
+    # Data entry mode setting 0x03 = X/Y increment
+    # IO.inspect("# Data entry mode setting 0x03 = X/Y increment")
+    send_command(state, 0x11, 0x03)
+
+    # Power on
+    # IO.inspect("# Power on")
+    send_command(state, 0x04)
+
+    # VCOM Register, 0x3c = -1.5v?
+    # IO.inspect("# VCOM Register, 0x3c = -1.5v?")
+    send_command(state, 0x2C, 0x3C)
+    send_command(state, 0x3C, 0x00)
+
+    # Always black border
+    # IO.inspect("# Always black border")
+    send_command(state, 0x3C, 0x00)
+
+    # Set voltage of VSH and VSL on Yellow device
+    if state.color == :yellow do
+      send_command(state, 0x04, 0x07)
+    end
+
+    # Set LUTs
+    # IO.inspect("# Set LUTs")
+    send_command(state, 0x32, get_luts(state.color))
+
+    # Set RAM X Start/End
+    # IO.inspect("# Set RAM X Start/End")
+    send_command(state, 0x44, :binary.list_to_bin([0x00, trunc(state.columns / 8) - 1]))
+
+    # Set RAM Y Start/End
+    # IO.inspect("# Set RAM Y Start/End")
+    send_command(state, 0x45, :binary.list_to_bin([0x00, 0x00] ++ packed_height))
+
+    # 0x24 == RAM B/W, 0x26 == RAM Red/Yellow/etc
+    for data <- [{0x24, buffer_a}, {0x26, buffer_b}] do
+      {cmd, buffer} = data
+
+      # Set RAM X Pointer start
+      # IO.inspect("# Set RAM X Pointer start")
+      send_command(state, 0x4E, 0x00)
+
+      # Set RAM Y Pointer start
+      # IO.inspect("# Set RAM Y Pointer start")
+      send_command(state, 0x4F, <<0x00, 0x00>>)
+      # IO.inspect("# Buffer thing")
+      send_command(state, cmd, buffer)
+    end
+
+    # Display Update Sequence
+    # IO.inspect("# Display Update Sequence")
+    send_command(state, 0x22, 0xC7)
+
+    # Trigger Display Update
+    # IO.inspect("# Trigger Display Update")
+    send_command(state, 0x20)
+
+    # Wait Before Deep Sleep
+    :timer.sleep(50)
+    busy_wait(state)
+
+    # Enter Deep Sleep
+    # IO.inspect("# Enter deep sleep")
+    send_command(state, 0x10, 0x01)
+  end
+
+  def pixels_to_bytestring(state = %State{}, color_value, match, no_match) do
+    rotation = state.rotation / 90
+
+    {order, outer_from, outer_to, inner_from, inner_to} =
+      case rotation do
+        -1.0 -> {:x_outer, state.width - 1, 0, 0, state.height - 1}
+        1.0 -> {:x_outer, 0, state.width - 1, state.height - 1, 0}
+        -2.0 -> {:y_outer, state.width - 1, 0, state.height - 1, 0}
+        _ -> {:y_outer, 0, state.height - 1, 0, state.width - 1}
+      end
+
+    for i <-
+          Enum.flat_map(outer_from..outer_to, fn i ->
+            Enum.map(inner_from..inner_to, fn j ->
+              key =
+                case order do
+                  :x_outer -> {i, j}
+                  :y_outer -> {j, i}
+                end
+
+              case state.pixels[key] do
+                ^color_value -> match
+                _ -> no_match
+              end
+            end)
+          end),
+        do: <<i::1>>,
+        into: <<>>
```

Comment:
You should be able to rewrite this as follows:

```elixir
for i <- outer_from..outer_to,
    j <- inner_from..inner_to,
    into: <<>> do
  key =
    # ...
end
```

Reply:
I'm a bit uncertain what behavior I could expect from this. Will it do the same thing as a flatmap. I'm guessing for is a macro but damn if I know what it actually expands into :P